test/Microsoft.Azure.SignalR.Tests/TestServiceConnection.cs

```diff
@@ -14,19 +15,27 @@ internal sealed class TestServiceConnection : IServiceConnection
         public Task ConnectionInitializedTask => Task.CompletedTask;
 
         private readonly bool _throws;
-        public TestServiceConnection(ServiceConnectionStatus status = ServiceConnectionStatus.Connected, bool throws = false)
+        private readonly TaskCompletionSource<object> _writeAsyncTcs = null;
+
+        public event Action<StatusChange> ConnectionStatusChanged;
+
+        public TestServiceConnection(ServiceConnectionStatus status = ServiceConnectionStatus.Connected, bool throws = false, TaskCompletionSource<object> writeAsyncTcs = null)
         {
             Status = status;
             _throws = throws;
+            _writeAsyncTcs = writeAsyncTcs;
         }
 
         public Task StartAsync(string target = null)
         {
+            ConnectionStatusChanged?.Invoke(new StatusChange(ServiceConnectionStatus.Inited, Status));
```

Comment:
>Inited [](start = 85, length = 6)

why added an "Inited" status but only used in test? not Connecting?

Reply:
`inited` status is added for ServiceConnectionContainer to have an initial state `Inited` instead of `Disconnected` as the initial state, so that ServiceConnectionContainer can get notified when the inner server connections has status changed to `Disconnected`. And I am not able to reuse `Connecting` as for backward compatibility, the status of `ServiceConnectionContainer` should be considered as online until it starts the inner service connection.

Reply:
So with `inited` we explicitly defined a stage for the server connection between it is created to it is `connected`. I think it is clear, as for now, we consider the endpoint as `online` in this stage, which is when the app server starts. And later, we can easily change the endpoint status for this stage if there is a requirement.